bbb/meta-bbb/recipes-core/trebletrouble/files/src/tone.c

```diff
@@ -0,0 +1,255 @@
+#include <stdio.h>
+#include <malloc.h>
+#include <math.h>
+
+#include "libfft.h"
+#include "tone.h"
+
+#define FFT_SIZE 15
+
+/*
+ * The audio format will be PCM.
+ *  
+ * -------------Data to be determined ----------------
+ * numChannels -  1 for mono, 2 for stereo, etc...
+ * sampleRate - samples per second (44100 or 96000)
+ * bitsPerSample - 8 or 16 or 32. The higher, the better the resolution.
+ * subChunk2Size - size of the actual sound data.
+ * chunkSize - the size of the sound data + size of the header.
+ *  
+ */
+
+
+/* ------------------------------------------------------ [ Section: Endians ] */
```

Comment:
Perhaps it would be a good idea to move these endian operations into their own file?